src/secure_channel.rs

```diff
@@ -52,45 +50,81 @@ impl ResourceDescription {
     }
 }
 
-/// SecureChannel to connect with KBS
-pub struct SecureChannel {
-    /// Get Resource Service client.
-    client: Client,
-    kbc_name: String,
-    kbs_uri: String,
-}
-
 /// Types of the Client underlying a SecureChannel
+#[allow(dead_code)]
 enum Client {
+    /// Fake client which just return errors.
+    None,
+    #[cfg(feature = "keywrap-grpc")]
     /// Get Resource Service gRPC client
-    ServiceGPRC(GetResourceServiceClient<Channel>),
+    ServiceGPRC(GetResourceServiceClient<tonic::transport::Channel>),
+    #[cfg(feature = "keywrap-native")]
     /// Get Rserouce native AA client
-    NativeAA(AttestationAgent),
+    NativeAA(attestation_agent::AttestationAgent),
 }
 
 impl Client {
     /// Retrieves confidential resource
     async fn get_resource(
         &mut self,
-        kbc_name: &str,
-        kbs_uri: &str,
-        resource_description: String,
+        _kbc_name: &str,
+        _kbs_uri: &str,
+        _resource_description: String,
     ) -> Result<Vec<u8>> {
         match self {
+            Self::None => Err(anyhow!("no mechanism to fetch resources")),
```

Comment:
IMO a better way is to raise the error when create the instance of `Client`. We can leave as what this looks now, and I'd refactor the logic in another pr.